daemon/logger/gelf/gelf.go

```diff
@@ -0,0 +1,114 @@
+package gelf
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Graylog2/go-gelf/gelf"
+	"github.com/docker/docker/daemon/logger"
+	"github.com/docker/docker/pkg/parsers"
+)
+
+type GelfLogger struct {
+	writer   *gelf.Writer
+	metadata GelfMetadata
+}
+
+type GelfMetadata struct {
+	ContainerId string
+	ImageId     string
+	Entrypoint  string
+	Args        []string
+}
+
+func New(options []string, gelfMetadata GelfMetadata) (logger.Logger, error) {
+	host := "localhost"
+	port := "12201"
+
+	// parse the options
+	for _, option := range options {
+		key, val, err := parsers.ParseKeyValueOpt(option)
+		if err != nil {
+			return nil, err
+		}
+		key = strings.ToLower(key)
+		switch key {
+		case "gelf.host":
+			host = val
+		case "gelf.port":
+			port = val
+		}
+	}
+
+	gelfWriter, err := gelf.NewWriter(fmt.Sprintf("%s:%s", host, port))
+	if err != nil {
+		return nil, fmt.Errorf("Can not connect to GELF endpoint: %v", err)
+	}
+
+	return &GelfLogger{
+		writer:   gelfWriter,
+		metadata: gelfMetadata,
+	}, nil
+}
+
+func (s *GelfLogger) Log(msg *logger.Message) error {
+	// remove trailing and leading whitespace
```

Comment:
Leading whitespaces are only interesting in a terminal where you want to have some kind of formated output. When you read the logs in a web interface the whitespaces are just noise.